docs/proposals/security-context-constraints.md

```diff
@@ -0,0 +1,320 @@
+// be applied to a container.
+type SecurityContextConstraints struct {
+	TypeMeta   `json:",inline"`
+	ObjectMeta `json:"metadata,omitempty"`
+
+	// AllowPrivilegedContainer determines if a container can request to be run as privileged.
+	AllowPrivilegedContainer bool `json:"allowPrivilegedContainer,omitempty" description:"allow containers to run as privileged"`
+	// AllowedCapabilities is a list of capabilities that can be requested to add to the container.
+	AllowedCapabilities []CapabilityType `json:"allowedCapabilities,omitempty" description:"capabilities that are allowed to be added"`
+	// AllowHostDirVolumePlugin determines if the policy allow containers to use the HostDir volume plugin
+	AllowHostDirVolumePlugin bool `json:"allowHostDirVolumePlugin,omitempty" description:"allow the use of the host dir volume plugin"`
+	// SELinuxContext is the strategy that will dictate what labels will be set in the SecurityContext.
+	SELinuxContext SELinuxContextStrategyOptions `json:"seLinuxContext,omitempty" description:"strategy used to generate SELinuxOptions"`
+	// RunAsUser is the strategy that will dictate what RunAsUser is used in the SecurityContext.
+	RunAsUser RunAsUserStrategyOptions `json:"runAsUser,omitempty" description:"strategy used to generate RunAsUser"`
+
+	// The users who have permissions to use this security context constraints
+	Users []string `json:"users,omitempty" description:"users allowed to use this SecurityContextConstraints"`
+	// The groups that have permission to use this security context constraints
+	Groups []string `json:"groups,omitempty" description:"groups allowed to use this SecurityContextConstraints"`
+}
+
+// SELinuxContextStrategyOptions defines the strategy type and any options used to create the strategy.
+type SELinuxContextStrategyOptions struct {
+	// Type is the strategy that will dictate what SELinux context is used in the SecurityContext.
+	Type SELinuxContextStrategyType `json:"type,omitempty" description:"strategy used to generate the SELinux context"`
+
+	//TODO
+}
+
+// RunAsUserStrategyOptions defines the strategy type and any options used to create the strategy.
+type RunAsUserStrategyOptions struct {
+	// Type is the strategy that will dictate what RunAsUser is used in the SecurityContext.
+	Type RunAsUserStrategyType `json:"type,omitempty" description:"strategy used to generate RunAsUser"`
+	// UID is the user id that containers must run as.  Required for the MustRunAs strategy if not using
+	// namespace/service account allocated uids.
+	UID *int64 `json:"uid,omitempty" description:"the uid to always run as; required for MustRunAs"`
+	// AllocatedIDAnnotation provides an annotation that the strategy can look for if it should
+	// use a pre-allocated UID that exists on the namespace or service account.
+	AllocatedIDAnnotation string `json:"allocatedIDAnnotation,omitempty" description:"how the strategy can discover pre-allocated ids on the namespace or service account"`
```

Comment:
I'm a bit worried that this is over complicated - why does the scc need to know the annotation?  Seems like this is inverted - the admission controller that applies the scc should properly resolve that value onto the scc prior to execution.  I don't think it needs to be on the scc.

> On May 26, 2015, at 4:22 PM, Eric Tune <notifications@github.com> wrote:
> 
> In docs/proposals/security-context-constraints.md:
> 
> > +	// Type is the strategy that will dictate what SELinux context is used in the SecurityContext.
> > +	Type SELinuxContextStrategyType `json:"type,omitempty" description:"strategy used to generate the SELinux context"`
> > +
> > +	//TODO
> > +}
> > +
> > +// RunAsUserStrategyOptions defines the strategy type and any options used to create the strategy.
> > +type RunAsUserStrategyOptions struct {
> > +	// Type is the strategy that will dictate what RunAsUser is used in the SecurityContext.
> > +	Type RunAsUserStrategyType `json:"type,omitempty" description:"strategy used to generate RunAsUser"`
> > +	// UID is the user id that containers must run as.  Required for the MustRunAs strategy if not using
> > +	// namespace/service account allocated uids.
> > +	UID *int64 `json:"uid,omitempty" description:"the uid to always run as; required for MustRunAs"`
> > +	// AllocatedIDAnnotation provides an annotation that the strategy can look for if it should
> > +	// use a pre-allocated UID that exists on the namespace or service account.
> > +	AllocatedIDAnnotation string `json:"allocatedIDAnnotation,omitempty" description:"how the strategy can discover pre-allocated ids on the namespace or service account"`
> @bgrant0607 any feeling on having a FooAnnotation field on a message which is not itself a resource or subresource?
> 
> —
> Reply to this email directly or view it on GitHub.
>